pkg/user/tx_client.go

```diff
@@ -157,9 +158,16 @@ type TxClient struct {
 	// that was submitted to the chain
 	txTracker           map[string]txInfo
 	gasEstimationClient gasestimation.GasEstimatorClient
+	// trustedMode indicates whether to trust the consensus node for account information.
+	// When true (default), the client trusts the consensus node to provide accurate
+	// account numbers and sequences without additional verification.
+	// When false, additional verification steps would be performed (to be implemented).
+	trustedMode bool
 }
 
-// NewTxClient returns a new signer using the provided keyring
+// NewTxClient returns a new TxClient using the provided keyring.
+// By default, the client operates in trusted mode, which assumes the consensus
+// node is honest.
```

Comment:
[nit] it's not really by default because it's forced. There is no way to override the default.

```suggestion
// The client operates in trusted mode, which assumes the consensus
// node is honest.
```